ocean_lib/models/datatoken.py

```diff
@@ -311,6 +307,96 @@ def build_access_service(
             consumer_parameters=consumer_parameters,
         )
 
+    def dispense_and_order(
+        self,
+        consumer: str,
+        service_index: int,
+        provider_fees: dict,
+        transaction_parameters: dict,
+        consume_market_order_fee_address: Optional[str] = None,
+        consume_market_order_fee_token: Optional[str] = None,
+        consume_market_order_fee_amount: Optional[int] = 0,
+    ) -> str:
+        buyer_addr = (
+            transaction_parameters["from"].address
+            if hasattr(transaction_parameters["from"], "address")
+            else transaction_parameters["from"]
+        )
+
+        bal = Web3.fromWei(self.balanceOf(buyer_addr), "ether")
+        if bal < 1.0:
+            dispenser_addr = get_address_of_type(self.config_dict, "Dispenser")
+            from ocean_lib.models.dispenser import Dispenser  # isort: skip
+
+            dispenser = Dispenser(self.config_dict, dispenser_addr)
+
+            # catch key failure modes
+            st = dispenser.status(self.address)
+            active, allowedSwapper = st[0], st[6]
+            if not active:
+                raise ValueError("No active dispenser for datatoken")
+            if allowedSwapper not in [ZERO_ADDRESS, buyer_addr]:
+                raise ValueError(f"Not allowed. allowedSwapper={allowedSwapper}")
+
+            # Try to dispense. If other issues, they'll pop out
+            dispenser.dispense(
+                self.address, "1 ether", buyer_addr, transaction_parameters
+            )
+
+        return self.start_order(
+            consumer=ContractBase.to_checksum_address(consumer),
+            service_index=service_index,
+            provider_fees=provider_fees,
+            consume_market_order_fee_address=(
+                consume_market_order_fee_address or ZERO_ADDRESS
+            ),
+            consume_market_order_fee_token=(
+                consume_market_order_fee_token or ZERO_ADDRESS
+            ),
+            consume_market_order_fee_amount=consume_market_order_fee_amount,
+            transaction_parameters=transaction_parameters,
+        )
+
+    @enforce_types
+    def buy_from_exchange_and_order(
```

Comment:
I realized that a better label yet is "buy_DT_and_order". 
- more consistent with "buy_DT()"
- more precise. (Are you buying a DT or basetoken?)